backend/scripts/import-posts/map-airtable-data.js

```diff
@@ -0,0 +1,76 @@
+// https://stackoverflow.com/questions/52456065/how-to-format-and-validate-email-node-js/52456632
+const emailRegexp = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+const cleanEmail = (email) => {
+  if (!email) return false;
+  // a few with multiple emails, only take first
+  let [cleaned] = email.split(",");
+  // trim trailing whitespace & periods
+  cleaned = cleaned.trim();
+  cleaned = cleaned.replace(/^\.|\.$/g, "");
+  // some other non-emails (extra link, whatsapp), just skip these
+  return emailRegexp.test(cleaned) && cleaned;
+};
+
+module.exports = (post, sourcedBy) => {
+  /* eslint-disable prefer-const */
+  let {
+    android_app: playStore,
+    city,
+    country,
+    from_whom: [authorType] = ["Other"], // provided as array even though we only want [one]
+    ios_app: appStore,
+    post_description: content,
+    post_email: email,
+    post_language: language,
+    post_name: title,
+    post_objective: [objective],
+    post_type: types,
+    post_website: website,
+    share_with: [visibility] = ["worldwide"],
+    state,
+  } = post.fields;
+  /* eslint-enable prefer-const */
+
+  if (authorType === "Others") {
+    authorType = "Other";
+  } else if (authorType === "Non-Profit") {
+    authorType = "Non-profit";
+  }
```

Comment:
These small data adaptations, maybe we should discuss with Andrés and data collection team to use the same format as we expect it in the DB from now on and just do a "replace all" to change the old ones.
What do you think?

Reply:
I asked him. I think it's a very simple change to just change a select option in. Since we're doing the cleaning manually already, such custom code shouldn't really be needed in our script. The exception email, I asked Andres about it and it sounds like it would take a lot of time to through them all again.

Reply:
Some was cleaned but I added `validationSync` on each doc and found new issues. Neither `bulkWrite` nor `findOneAndUpdate` will validate based on the schema (although they will transform to lowercase, trim whitespace).  For things that I can fix in the script (i.e. it's obvious what the correct alternate value is) I don't want to force new work on the data collection team. I just need to update for user model v2 and then it should be good.